test/test-util.js

```diff
@@ -18,7 +18,32 @@ function isLoaded(node, nodeName, done) {
   });
 }
 
+function correctOutputReceived(flow, givenInput, expectedOutput, done) {
+  nodeTestHelper.load(flow.nodes, flow.flow, function() {
+    const inputNodeId = flow.flow[0].id;
+    const inputNode = nodeTestHelper.getNode(inputNodeId);
+
+    const outputNodeId = flow.flow[flow.flow.length - 1].id;
+    const outputNode = nodeTestHelper.getNode(outputNodeId);
+
+    outputNode.on('input', function(msg){
+      try {
+        assert.deepEqual(msg.payload, expectedOutput);
+        done();
+      }
+      catch(err){
+        done(err);
+      }
+      finally{
+        inputNode.shell.stop();
```

Comment:
Instead of doing `this.shell = shell` in `quantum-circuit.js`, import `PythonShell` in `test-util.js` and call `python.stop()` directly. You can see an example of where I've done this [here](https://github.com/louislefevre/node-red-contrib-quantum/blob/4e72d2d6d8d042c2b8dd3f9641fbcb72c6d80ec7/test/test-util.js#L36).

Reply:
This makes sense. Thank you.